node/src/daemon/setup_reporter.rs

```diff
@@ -801,16 +816,21 @@ impl ValueRetriever for MappingProtocol {
 
     fn computed_default(
         &self,
-        _bootstrapper_config: &BootstrapperConfig,
+        bootstrapper_config: &BootstrapperConfig,
         persistent_config: &dyn PersistentConfiguration,
         _db_password_opt: &Option<String>,
     ) -> Option<(String, UiSetupResponseValueStatus)> {
-        let persistent_config_value_opt = match persistent_config.mapping_protocol() {
-            Ok(protocol_opt) => protocol_opt,
-            Err(_) => None,
-        };
-        persistent_config_value_opt
-            .map(|protocol| (protocol.to_string().to_lowercase(), Configured))
+        match (
+            bootstrapper_config.mapping_protocol_opt,
+            persistent_config.mapping_protocol(),
+        ) {
+            (_, Err(e)) => todo!("Error retrieving mapping protocol from database: {:?}", e),
```

Comment:
You left an unimplemented arm in this match. Please drive it in by a new test.

Reply:
I decided to turn it into a `panic!` because of that card in the backlog somewhere about modifying `PersistentConfiguration` so that it panics rather than returning errors when there are database problems. The client of `PersistentConfiguration` can almost never do anything about database errors; therefore they all panic anyway; therefore why not move the panic to the `PersistentConfiguration` and save some code?